solidity/contracts/NetworkSettings.sol

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: SEE LICENSE IN LICENSE
+pragma solidity 0.6.12;
+
+import "./INetworkSettings.sol";
+import "./utility/Owned.sol";
+import "./utility/Utils.sol";
+
+/**
+ * @dev This contract maintains the network settings.
+ */
+contract NetworkSettings is INetworkSettings, Owned, Utils {
+    address private _networkFeeWallet;
+    uint32 private _networkFee;
+
+    // ensures that the fee is valid
+    modifier validFee(uint32 fee) {
+        _validFee(fee);
+        _;
+    }
+
+    // error message binary size optimization
+    function _validFee(uint32 fee) internal pure {
+        require(fee <= PPM_RESOLUTION, "ERR_INVALID_FEE");
+    }
+
+    /**
+     * @dev initializes a new NetworkSettings contract
+     *
+     * @param initialNetworkFeeWallet initial network fee wallet
+     * @param initialNetworkFee initial network fee in ppm units
+     */
+    constructor(address initialNetworkFeeWallet, uint32 initialNetworkFee) validAddress(initialNetworkFeeWallet) validFee(initialNetworkFee) public {
+        _networkFeeWallet = initialNetworkFeeWallet;
+        _networkFee = initialNetworkFee;
+    }
+
+    /**
+     * @dev returns the network fee parameters
+     *
+     * @return network fee wallet
+     * @return network fee in ppm units
+     */
+    function networkFeeParams() external view override returns (address, uint32) {
+        return (_networkFeeWallet, _networkFee);
+    }
+
+    /**
+     * @dev returns the network fee wallet
+     *
+     * @return network fee wallet
+     */
+    function networkFeeWallet() external view override returns (address) {
+        return _networkFeeWallet;
+    }
+
+    /**
+     * @dev returns the network fee
```

Comment:
add a clause that describes what the fee is - fee that's taken as a portion from all swaps in the network

Reply:
Added at the top of the file (since the term 'network fee' appears in several places).